Resources/public/js/models/ez-locationmodel.js

```diff
@@ -245,7 +250,31 @@ YUI.add('ez-locationmodel', function (Y) {
                 query,
                 callback
             );
-        }
+        },
+
+        /**
+         * Overrides the RestModel implementation to also deal with the `path` attribute
+         *
+         * @method toJSON
+         * @return {Object}
+         */
+        toJSON: function () {
+            var attrs = Y.eZ.RestModel.prototype.toJSON.call(this),
+                path = [];
+
+            if (attrs.path) {
+                Y.Object.each(attrs.path, function (value, key) {
```

Comment:
when filled, `path` is an array, so a Y.Array method would more suited also since you want to create an array you can write:
```js
attrs.path = Y.Array.map(attrs.path, function (value) {
    return value.toJSON();
});
```